packages/node/src/channels.ts

```diff
@@ -0,0 +1,328 @@
+    initialAppsNonce: Nonce
+  ): {
+    [assetType: number]: FreeBalance;
+  } {
+    // TODO: extend to all asset types
+    const ethFreeBalance = new FreeBalance(
+      multisigOwners[0],
+      ethers.utils.bigNumberify("0"),
+      multisigOwners[1],
+      ethers.utils.bigNumberify("0"),
+      0,
+      0,
+      0,
+      initialAppsNonce
+    );
+    return {
+      [AssetType.ETH]: ethFreeBalance
+    };
+  }
+}
+
+/**
+ * Note: this class itelf does not hold any meaningful state either.
+ * It encapsulates the operations performed on relevant appInstances and
+ * abstracts the persistence to the store service.
+ */
+export class Channels {
+  /**
+   * A convenience lookup table from a set of owners to multisig address.
+   */
+  private readonly ownersToMultisigAddress = {};
+
+  /**
+   * A convenience lookup table from appInstance ID to multisig address.
+   */
+  private readonly appInstanceIdToMultisigAddress = {};
+
+  /**
+   * @param selfAddress The address of the account being used with the Node.
+   * @param store
+   * @param multisigKeyPrefix The prefix to add to the key being used
+   *        for indexing multisig addresses according to the execution
+   *        environment.
+   */
+  constructor(
+    public readonly selfAddress: Address,
+    private readonly store: IStoreService,
+    private readonly multisigKeyPrefix: string
+  ) {}
+
+  /**
+   * Called when a new multisig is created for a set of owners.
+   * @param multisigAddress
+   * @param multisigOwners
+   * @param freeBalances
+   */
+  async createMultisig(params: Node.CreateMultisigParams): Promise<Address> {
+    const multisigAddress = Channels.getMultisigAddress(params.owners);
+    const channel: Channel = new Channel(multisigAddress, params.owners);
+    const ownersHash = Channels.canonicalizeAddresses(params.owners);
+    this.ownersToMultisigAddress[ownersHash] = multisigAddress;
+    await this.save(channel);
+    return multisigAddress;
+  }
+
+  async addMultisig(multisigAddress: Address, owners: Address[]) {
+    const channel = new Channel(multisigAddress, owners);
+    const ownersHash = Channels.canonicalizeAddresses(owners);
+    this.ownersToMultisigAddress[ownersHash] = multisigAddress;
+    await this.save(channel);
+  }
+
+  /**
+   * Returns a JSON object with the keys being the multisig addresses and the
+   * values being objects reflecting the channel schema described above.
+   */
+  async getAllChannels(): Promise<object> {
+    const channels = await this.store.get(this.multisigKeyPrefix);
+    if (!channels) {
+      console.log("No channels exist yet");
+      return {};
+    }
+    return channels;
+  }
+
+  private async getChannelFromPeerAddress(
+    peerAddress: Address
+  ): Promise<Channel> {
+    const owners = Channels.canonicalizeAddresses([
+      this.selfAddress,
+      peerAddress
+    ]);
+    const multisigAddress = this.ownersToMultisigAddress[owners];
+    const channel = await this.store.get(
```

Comment:
@cf19drofxots Should we maybe abstract this into a `getChannelFromStore()` function?